Automattic/jetpack
# AI Chat: Add nudges so that AI Chat is properly configured

For the time being, this assumes that Jetpack search plan is ALL that you need to activate the feature. In the future, this probably should change. However, with delays in the Jetpack Metered Billing product, I am treating this as part of Jetpack Search for now.


<img width="1025" alt="Zrzut ekranu 2023-09-14 o 11 26 06" src="https://github.com/Automattic/jetpack/assets/3775068/38da6279-c14c-4b50-8123-59d1e413a799">


Fixes https://github.com/Automattic/jetpack/issues/32955

## Proposed changes:
* Pass in a new parameter in `window.Jetpack_Editor_Initial_State.search`
* Add a nudge to check status of Jetpack connection
* Add a nudge to check if Jetpack search is enabled


## Does this pull request change what data or activity we track or use?
NO

## Testing instructions:


1. Get a Jetpack site
2. `define( 'JETPACK_BLOCKS_VARIATION', 'beta' );`
3. Publish a few posts
4. Insert the block on a page
5. **Observe the prompt to connect Jetpack**
6. Connect to WPCOM.
7.  **Observe the prompt to enable Jetpack search.**
10. Enable it
11. Observe all prompts gone.
12. publish the page from 4
13. Open the page on the frontend
14. Ask a question relating to your published posts
15. Get a WPCOM site, sandbox public-api AND site domain
16. `bin/jetpack-downloader test jetpack chat-with-site/wpcom`
4. Publish a few posts
5. Insert the block on a page
7.  **Observe the prompt to enable Jetpack search.**
10. Enable it
11. Observe all prompts gone.
12. Publish the page with block
13. Open the page on the frontend
14. Ask a question relating to your published posts
